astropy/modeling/fitting.py

```diff
@@ -729,11 +729,20 @@ def _wrap_deriv(params, model, weights, x, y, z=None):
             weights = 1.0
 
         if any(model.fixed.values()) or any(model.tied.values()):
-
+            # update the parameters with the current values from the fitter
+            _fitter_to_model_params(model, params)
             if z is None:
-                full_deriv = np.ravel(weights) * np.array(model.fit_deriv(x, *model.parameters))
+                full = np.ravel(weights) * np.array(model.fit_deriv(x, *model.parameters))
+                if not model.col_fit_deriv:
+                    full_deriv = np.ravel(weights) * full.T
```

Comment:
@nden - ``full_deriv`` will include ``np.ravel(weights)`` twice - is that normal? (that appears to be a change from before)

Reply:
@astrofrog Good catch! That would have applied weights twice. Thanks.